tensorboard/webapp/app_routing/effects/app_routing_effects.ts

```diff
@@ -181,12 +182,31 @@ export class AppRoutingEffects {
     .pipe(
       delay(0),
       map(() => {
+        const namespaceId: string | undefined =
+          this.location.getHistoryState()?.namespaceId;
+
+        const namespaceUpdate: NamespaceUpdate =
+          namespaceId === undefined
+            ? // There is no namespace id in the browser history entry. This is,
+              // therefore, some sort of new navigation to the app. Set options
+              // so that a new namespace id is set downstream.
+              {
+                option: NamespaceUpdateOption.NEW,
+              }
+            : // There is a namespace id in the browser history entry. This is,
+              // therefore, a page reload. Set options so that the namespace id
+              // is reused downstream.
+              {
+                option: NamespaceUpdateOption.FROM_HISTORY,
+                namespaceId: namespaceId,
+              };
+
         return {
           pathname: this.location.getPath(),
           options: {
             browserInitiated: true,
             replaceState: true,
-            namespaceUpdateOption: NamespaceUpdateOption.UNCHANGED,
+            namespaceUpdate,
```

Comment:
Note that I had to modify the NavigationOption typing so that I could construct the object this way without the compiler complaining. (Where "this way" is having a variable namespaceUpdate while other options -- browserInitiated and replaceState -- are statically set).